### ISSUE TITLE
[1.4] Asset Loading Bug; Poor modded error handling results in null vanilla textures

### ISSUE THREAD

**Issue by Solxanich** [tModLoader/tModLoader#1373]
### Description
Direwolf: If a mod crashes on loading assets (invalid texture path), it interferes with other texture loading

### Steps to reproduce
Direwolf: so basically, my setup is 1.4 branch, ClickerClass, and CheatSheet, both mods work together and in isolation
I added an item to ClickerClass and overridden the Texture property and gave it an invalid string
When I build the mod, and it loads the mod, it ends up throwing an error window first (saying relogic/assets/blabla content not found, pointing out the invalid path), clicking OK will then throw the usual "main menu error" window showing the same error as the window, and saying that ClickerClass is the problem. Then, the game disables the mod and loads all other mods. Here's the fun part. CheatSheet accesses textures from vanilla for its recipe browser UI. Those textures are now null (accessed for the first time in PostAddRecipes), meaning it throws, and causes alot of other issues along the way. (Screenshot reference 1 and 2)

The fun continues, if you restart the game, you enable ClickerClass (the version that got disabled because of an error),reload mods, and it will actually load in! No CheatSheet crash. Once I get into the game, and try looking for the item with the invalid asset in the item browser of CheatSheet, I get a crash from CheatSheet saying that the item texture is null (Screenshot reference 3)
1 and 2
![image](https://user-images.githubusercontent.com/59670736/111918829-9660b580-8a4c-11eb-9a8f-86a8024f5da3.png)
![image](https://user-images.githubusercontent.com/59670736/111918834-9b256980-8a4c-11eb-9813-b268b5ee0f52.png)

3 (5144 is that items type, accessed in TextureAssets.Item)
![image](https://user-images.githubusercontent.com/59670736/111918846-a082b400-8a4c-11eb-83eb-2e6239d0e71a.png)

bonus with (misleading) error after building (error does not appear on re-enabling it again after rebooting)
![image](https://user-images.githubusercontent.com/59670736/111918855-a7112b80-8a4c-11eb-9eaa-99196f1ef035.png)

tl;dr fix asset loading please :pleading_face:
I tried to debug it, but since its spread across threads and relogic stuff, I have no clue how to properly tackle this as my knowledge of AssetRepository is zero

### Expected behavior 
Asset should fail to load, toss appropriate error, and other mods/vanilla should load fine.

**Comment by Chicken-Bones** [comment]
Should definitely throw rather than returning null, though caching textures is not recommended at all.

**Comment by JavidPack** [comment]
I'm guessing this is fixed by now. We have code to fallback to a 1x1 texture and there have been many other changes over the years. If anyone has a similar issue and can provide steps we can reproduce, feel free to make a new issue.
